Sources/App/Controllers/PackageController+routes.swift

```diff
@@ -368,30 +368,28 @@ enum PackageController {
 
         // For repositories that have no README file at all.
         guard let readmeHtmlUrl = pkg.repository?.readmeHtmlUrl else {
-            return PackageReadme.View(model: .init(url: nil,
-                                                   repositoryOwner: nil,
-                                                   repositoryName: nil,
-                                                   defaultBranch: nil,
-                                                   readme: nil)).document()
+            return PackageReadme.View(model: .noReadme).document()
         }
 
         do {
             let readme = try await Current.fetchS3Readme(req.client, owner, repository)
+            guard let owner = pkg.repository?.owner,
+                  let repo = pkg.repository?.name,
+                  let branch = pkg.repository?.defaultBranch
+            else {
+                return PackageReadme.View(model: .cacheLookupFailed(url: readmeHtmlUrl)).document()
```

Comment:
I'm using `cacheLookupFailed` here, because it kinda is (we can't look up the cache without them) - although this could never really happen. The problem (that perhaps we could fix one day) is that `owner`, `repo`, and `branch` could be `nil` technically but never really are. That's due to how the fields are modelled in the db, which we could address ultimately.